### PR TITLE
fix: 2FA bug fixes

### PR DESCRIPTION
**Pull request by BeeMargarida**
### Details

Fix several small issues related with 2FA.

1. Missing Download translation in Spanish - https://expensify.slack.com/archives/C049HHMV9SM/p1684406965776349
2. Two-factor name is inconsistent https://expensify.slack.com/archives/C049HHMV9SM/p1684406403576119
3. <del>Offline Indicator displayed twice https://expensify.slack.com/archives/C049HHMV9SM/p1684408727072029</del>
   - This will be fixed in another issue, since it's present in other places - https://github.com/Expensify/App/issues/19187#issuecomment-1554306287 
4. Here is your magic code page appears https://expensify.slack.com/archives/C049HHMV9SM/p1684405641601719
5. Error message persists https://expensify.slack.com/archives/C049HHMV9SM/
6. View disrupted on small screen devices https://expensify.slack.com/archives/C049HHMV9SM/p1684392775324939
 

### Fixed Issues

$ https://github.com/Expensify/App/issues/19187
https://github.com/Expensify/App/issues/19187#issuecomment-1554254779

### Tests

Problem 1:
1. Sign in and enter the app
2. Change language to Spanish
3. Go to Seguridad -> Autenticación de dos factores 
4. See that the button says 'Descargar' instead of Download

Problem 2:
1. Navigate to Sign in page
3. Enter email & click on continue button
4. Open email, and copy magic sign in link
7. Paste magic sign-in link in a new tab & press enter
8. Observe that it says 'Two-factor authentication required' instead of 'Two factor authentication required'

Problem 4:
1. Go to Sign in page and enter the **email of an account with 2FA enabled** and click next
2. Open received email, and copy magic sign in link (check that this is using staging URL, if not change it)
3. Paste this link in a new tab & press enter
4. Observe that 'Two-factor authentication required' page is displayed
5. Go to the tab where you were signing in, which should have moved on to 2FA code
6. Enter the 2FA code
7. Verify that user is signed in
8. Shift to the tab where 'Two-factor authentication required' page is displayed
9. Observe that it should now show 'Abracadabra, you are signed in!'

Problem 5:
1. Go to the Settings > Security > Two-factor authentication
2. Copy to clipboard and click "Next"
3. Enter an invalid 6 digit number and notice an error message shows up
4. Click the cancel icon and close the side panel
5. Click on settings > security > 2FA again
6. Notice a new code appears
7. Click on copy and click next
8. Observe that the old error message does not appear

Problem 6:
1. In a small screen device (Samsung Galaxy S5 - available in browser dev tools - or in a device with 720 x 1640 Pixels (~264 PPI)) open the app/browser page
2. Go to the Settings > Security > Two-factor authentication
3. Verify that the codes appear in two columns and do not overlap the buttons

- [x] Verify that no errors appear in the JS console

### Offline tests

Do not apply for these specific issues, the only one that should affect won't be solved in this PR.

### QA Steps

Problem 1:
1. Sign in and enter the app
2. Change language to Spanish
3. Go to Seguridad -> Autenticación de dos factores 
4. See that the button says 'Descargar' instead of Download

Problem 2:
1. Navigate to Sign in page
3. Enter email & click on continue button
4. Open email, and copy magic sign in link
7. Paste magic sign-in link in a new tab & press enter
8. Observe that it says 'Two-factor authentication required' instead of 'Two factor authentication required'

Problem 4:
1. Go to Sign in page and enter the **email of an account with 2FA enabled** and click next
2. Open received email, and copy magic sign in link (locally this link needs to be switched to use localhost)
3. Paste this link in a new tab & press enter
4. Observe that 'Two-factor authentication required' page is displayed
5. Go to the tab where you were signing in, which should have moved on to 2FA code
6. Enter the 2FA code
7. Verify that user is signed in
8. Shift to the tab where 'Two-factor authentication required' page is displayed
9. Observe that it should now show 'Abracadabra, you are signed in!'

Problem 5:
1. Go to the Settings > Security > Two-factor authentication
2. Copy to clipboard and click "Next"
3. Enter an invalid 6 digit number and notice an error message shows up
4. Click the cancel icon and close the side panel
5. Click on settings > security > 2FA again
6. Notice a new code appears
7. Click on copy and click next
8. Observe that the old error message does not appear

Problem 6:
1. In a small screen device (Samsung Galaxy S5 - available in browser dev tools - or in a device with 720 x 1640 Pixels (~264 PPI)) open the app/browser page
2. Go to the Settings > Security > Two-factor authentication
3. Verify that the codes appear in two columns and do not overlap the buttons

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

Problem 1:
image

Problem 2:
image

Problem 4:

https://github.com/Expensify/App/assets/25725586/2e656eee-29e8-464e-ac9f-343010688d9e

Problem 5 and 6:

https://github.com/Expensify/App/assets/25725586/1bb5fe4a-ac3c-4d49-8061-e0e9c1e96ef2

</details>

<details>
<summary>Mobile Web - Chrome</summary>

Problem 6:

image

image

</details>

<details>
<summary>Mobile Web - Safari</summary>

Problem 6:

image

</details>

<details>
<summary>Desktop</summary>

image

</details>

<details>
<summary>iOS</summary>

Problem 6: 

image

</details>

<details>
<summary>Android</summary>

Problem 6:

image

</details>

### CODE REVIEW

File: src/pages/ValidateLoginPage/index.website.js [hunk 1]
```diff
@@ -117,14 +117,14 @@ class ValidateLoginPage extends Component {
     }
 
     render() {
-        const isTfaRequired = lodashGet(this.props, 'account.requiresTwoFactorAuth', false);
+        const is2FARequired = lodashGet(this.props, 'account.requiresTwoFactorAuth', false);
         const isSignedIn = Boolean(lodashGet(this.props, 'session.authToken', null));
         return (
             <>
                 {this.getAutoAuthState() === CONST.AUTO_AUTH_STATE.FAILED && <ExpiredValidateCodeModal />}
```
Comment (fabioh8010) [inline src/pages/ValidateLoginPage/index.website.js:124]: We have 5 calls in sequence to a method which will always give the same result during a `render`. Maybe we could extract `this.getAutoAuthState()` to a variable `const autoAuthState = this.getAutoAuthState();` and use it.

File: src/pages/settings/Security/TwoFactorAuth/CodesPage.js [hunk 2]
```diff
@@ -73,7 +75,7 @@ function CodesPage(props) {
                     <View style={styles.mv3}>
                         <Text>{props.translate('twoFactorAuth.codesLoseAccess')}</Text>
                     </View>
-                    <View style={styles.twoFactorAuthCodesBox}>
+                    <View style={[styles.twoFactorAuthCodesBox, {paddingHorizontal: props.isSmallScreenWidth ? 40 : 60}]}>
```
Comment (thiagobrez) [inline src/pages/settings/Security/TwoFactorAuth/CodesPage.js:78]: I think you should use it like this, to keep the project pattern:

```suggestion
                    <View style={[styles.twoFactorAuthCodesBox, props.isSmallScreenWidth ? styles.ph10 : styles.ph15]}>
```

And you'll need to create `styles.ph15`

**Review by MonilBhavsar — COMMENTED** [review]
Looks good to me 👍  Thanks!
@mananjadhav would like to get your review here and finish the checklist

**Review by mananjadhav — DISMISSED** [review]
Code changes are fine, I'll test on all platforms and update the checklist.

**Comment by mananjadhav** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android / native
    - [ ] Android / Chrome
    - [ ] iOS / native
    - [ ] iOS / Safari
    - [ ] MacOS / Chrome / Safari
    - [ ] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

web-2fa-translation-pr-1


</details>

<details>
<summary>Mobile Web - Chrome</summary>

mweb-2fa-code-column-pr-6

mweb-chrome-translation


</details>

<details>
<summary>Mobile Web - Safari</summary>

mweb-safari-translation

</details>

<details>
<summary>Desktop</summary>

desktop-2fa-translation

</details>

<details>
<summary>iOS</summary>



</details>

<details>
<summary>Android</summary>



</details>

**Comment by mananjadhav** [comment]
@MonilBhavsar @BeeMargarida Is the Two Factor Screen loading for you on Android and iOS?

On Android it is abruptly crashing for me without any errors. 

https://github.com/Expensify/App/assets/3069065/bb81121f-1e45-40db-862d-c197902ee55e



On iOS I get an exception first, and if I dismiss I see the screen.

image

Also on iOS I am unable to enter the 2fa code.

https://github.com/Expensify/App/assets/3069065/2f063260-8301-4317-af6c-da9b3c5f524a

**Comment by MonilBhavsar** [comment]
I'm seeing it too 😭 
@BeeMargarida or @thiagobrez if you could please take a look

**Comment by thiagobrez** [comment]
@MonilBhavsar Looking into it now!

**Comment by thiagobrez** [comment]
@mananjadhav @MonilBhavsar 

>On Android it is abruptly crashing for me without any errors.

>On iOS I get an exception first, and if I dismiss I see the screen.

These two ⬆️  were caused by a little issue, incorrectly setting a style. I'm contacting Ana to get permissions to push to her branch.

>Also on iOS I am unable to enter the 2fa code.

This one is a [known bug](https://github.com/Expensify/App/issues/18218), unrelated to 2-FA. Happens only on iOS 16.4, on simulator, in dev mode. See the related issue

**Comment by thiagobrez** [comment]
Should be good now 👌🏻

**Comment by mananjadhav** [comment]
Thanks @thiagobrez. I am afk, I'll test this within an 1-2 hours.

**Review by mananjadhav — DISMISSED** [review]
@MonilBhavsar I've finished the checklist. The android tests work fine now. 

I've bypassed the iOS issue by pasting the code, instead of typing in.

**Comment by MonilBhavsar** [comment]
@thiagobrez I see the view being disrupted bug on samsung galaxy fold. I think we still need to fix it. Could you please take look, thanks!

Screenshot 2023-05-29 at 9 38 06 PM

**Comment by mananjadhav** [comment]
Ohh yeah 🤦. Thanks for catching @MonilBhavsar

**Comment by thiagobrez** [comment]
Great catch @MonilBhavsar, thanks! I didn't know it was supported, I'll keep an eye next time 🙇🏻 

I added a new breakpoint for extra small screens like this. I'm using `< 320` since this is the smallest preset breakpoint on Chrome. Let me know what you think :)

Screenshot 2023-05-30 at 09 34 30

**Comment by MonilBhavsar** [comment]
Thanks for fixing this! The view looks good now, but looks like user can't see continue button. I wonder should we use ScrollView in codespage

File: ios/Podfile.lock
Comment (MonilBhavsar) [inline ios/Podfile.lock:1]: Thanks! Is this file really needed?

**Review by mananjadhav — APPROVED** [review]
Just tested the CodePages ScrollView.

File: src/styles/styles.js [hunk 3]
```diff
@@ -2127,14 +2127,26 @@ const styles = {
...
-    twoFactorAuthCodesBox: {
-        alignItems: 'center',
-        justifyContent: 'center',
-        backgroundColor: themeColors.highlightBG,
-        paddingVertical: 28,
-        paddingHorizontal: 60,
-        borderRadius: 16,
-        marginTop: 32,
+    twoFactorAuthCodesBox: ({isExtraSmallScreenWidth, isSmallScreenWidth}) => {
+        let paddingHorizontal = styles.ph15;
```
Comment (MonilBhavsar) [inline src/styles/styles.js:2131]: @shawnborton buddy check please for these new style

**Review by MonilBhavsar — APPROVED** [review]
Thanks! Looks good to me

**Review by MonilBhavsar — APPROVED** [review]
Thanks! Looks good to me

File: src/styles/styles.js (hunk 3 above)
Comment (shawnborton) [reply src/styles/styles.js:2131]: What exactly am I reviewing here? Just the screenshots from the first comment?

Comment (MonilBhavsar) [reply src/styles/styles.js:2131]: sorry, Nothing has changed much in the view. I wanted to get an eye on padding prooerty we added to support view for extra small devices.

Comment (shawnborton) [reply src/styles/styles.js:2131]: Ah got it. Can you show me screenshots of what this looks like in practice please? Thanks!

Comment (thiagobrez) [reply src/styles/styles.js:2131]: @shawnborton Here you go:

<details>
<summary>
Desktop - using padding from Figma: `styles.ph15`
</summary>

</details>

<details>
<summary>
Small screen width (<= 800px) - decreased padding until it doesn't break: `styles.ph10`
</summary>

</details>

<details>
<summary>
Extra small screen width (<= 320px) - decreased padding until it doesn't break: `styles.ph4`
</summary>

</details>

Comment (shawnborton) [reply src/styles/styles.js:2131]: Okay cool, that works for me!

Comment (MonilBhavsar) [reply src/styles/styles.js:2131]: Thanks shawn!

**Comment by cristipaval** [comment]
Heads up, this PR added [this regression](https://github.com/Expensify/App/issues/20136)

File: src/pages/ValidateLoginPage/index.website.js [hunk 4]
```diff
@@ -117,20 +117,21 @@ class ValidateLoginPage extends Component {
...
         return (
             <>
-                {this.getAutoAuthState() === CONST.AUTO_AUTH_STATE.FAILED && <ExpiredValidateCodeModal />}
-                {this.getAutoAuthState() === CONST.AUTO_AUTH_STATE.JUST_SIGNED_IN && (!isTfaRequired || isSignedIn) && <AbracadabraModal />}
-                {this.getAutoAuthState() === CONST.AUTO_AUTH_STATE.JUST_SIGNED_IN && isTfaRequired && !isSignedIn && <TfaRequiredModal />}
-                {this.getAutoAuthState() === CONST.AUTO_AUTH_STATE.NOT_STARTED && (
+                {currentAuthState === CONST.AUTO_AUTH_STATE.FAILED && <ExpiredValidateCodeModal />}
+                {currentAuthState === CONST.AUTO_AUTH_STATE.JUST_SIGNED_IN && is2FARequired && !isSignedIn && <TfaRequiredModal />}
+                {currentAuthState === CONST.AUTO_AUTH_STATE.NOT_STARTED && isSignedIn && <AbracadabraModal />}
+                {currentAuthState === CONST.AUTO_AUTH_STATE.NOT_STARTED && !isSignedIn && (
```
[the remark this answers is not in the document]
Comment (BeeMargarida) [reply src/pages/ValidateLoginPage/index.website.js:128]: @s77rt  Yes, it came from this [commit](https://github.com/Expensify/App/pull/19260/commits/4da2f319f04950908af5a04c434aba1510043252) to fix this bug mentioned in this [slack thread](https://expensify.slack.com/archives/C049HHMV9SM/p1684405641601719).

Before, it was necessary because the check for `AbracadabraModal` (now `JustSignedInModal`) used `NOT_STARTED` state (this is because the state was cleaned on startup - mentioned in the [proposal](https://github.com/Expensify/App/issues/19187#issuecomment-1554254779)), so we needed to differentiate between that one and the one for `ValidateCodeModal`. At the time, the only difference was that for the abracadabra, the `isSignedIn` was true and false for the `ValidateCodeModal`.

[the remark this answers is not in the document]
Comment (s77rt) [reply src/pages/ValidateLoginPage/index.website.js:128]: @BeeMargarida Thank you!

File: src/pages/settings/Security/TwoFactorAuth/CodesPage.js [hunk 5]
```diff
@@ -64,65 +67,67 @@ function CodesPage(props) {
...
+                        </View>
+                    </Section>
+                    <FixedFooter style={[styles.twoFactorAuthFooter]}>
+                        <Button
+                            success
+                            text={props.translate('common.next')}
+                            onPress={() => Navigation.navigate(ROUTES.SETTINGS_2FA_VERIFY)}
+                            isDisabled={isNextButtonDisabled}
+                        />
+                    </FixedFooter>
```
Comment (Santhosh-Sellavel) [inline src/pages/settings/Security/TwoFactorAuth/CodesPage.js:129]: Seems we accidentally included Fixed inside scroll containers and that led to this regression https://github.com/Expensify/App/issues/20138
